sd_dynamic_prompts/ui/prompt_writer.py

```diff
@@ -0,0 +1,57 @@
+from __future__ import annotations
+
+import csv
+from pathlib import Path
+
+from dynamicprompts import constants
+
+
+class PromptWriter:
+    def __init__(self):
+        self.reset()
+        self._enabled = False
+
+    def reset(self):
+        self._already_saved = False
+        self._positive_prompt = ""
+        self._negative_prompt = ""
+        self._positive_prompts = []
+        self._negative_prompts = []
+
+    @property
+    def enabled(self) -> bool:
+        return self._enabled
+
+    def set_enabled(self, value: bool) -> None:
+        self._enabled = value
+
+    def set_data(
+        self,
```

Comment:
Could this be made kwarg-only? It's otherwise easy to pass args in the incorrect order.